funsor/ops/op.py

```diff
@@ -170,7 +189,19 @@ def __call__(self, *args, **kwargs):
 
         # Dispatch.
         fn = cls.dispatcher.partial_call(*args[: cls.arity])
-        return fn(*args, **kwargs)
+        if instrument.DEBUG:
+            print(instrument.get_indent() + repr(self))
+            instrument.STACK_SIZE += 1
+            try:
+                result = fn(*args, **kwargs)
+            finally:
+                instrument.STACK_SIZE -= 1
+        else:
+            result = fn(*args, **kwargs)
+
+        if _TRACE is not None:
+            _TRACE.setdefault(id(result), (result, self, raw_args))
```

Comment:
@eb8680 note we can't use your "don't trace inside an op" trick because backend ops can occur inside irrelevant Funsor ops

Reply:
Not sure I understand this, can you give an example? Why would the compiler need to know what happens inside an op?

Reply:
Sure, I've added [two](https://github.com/pyro-ppl/funsor/pull/557/files#diff-7ef4fc236f9d538cec3c7d66cfbaea8f6e6893a40acb42c582bea0807480e5e5R63) [examples](https://github.com/pyro-ppl/funsor/pull/557/files#diff-7ef4fc236f9d538cec3c7d66cfbaea8f6e6893a40acb42c582bea0807480e5e5R80) in the tests.  A function that inputs and outputs backend arrays may use Funsors internally, triggering irrelevant Funsor ops inside of which relevant backend ops are triggered.  This is our usage pattern in `AutoGaussian`.

Ok... I got your idea to work by filtering out irrelevant ops.

Reply:
There is no need for workaround, I got your idea to work by filtering out irrelevant ops.